src/Config/MosaicConfigV2.ts

```diff
@@ -0,0 +1,89 @@
+import * as path from "path";
+import Directory from "../Directory";
+import Logger from "../Logger";
+import * as fs from "fs-extra";
+
+/**
+ * Holds the config of mosaic chains of a specific origin chain.
+ */
+export default class MosaicConfig {
+    public originChain: OriginChain;
+    public auxiliaryChains: AuxiliaryChain[];
```

Comment:
I think we said this should be a mapping from auxiliary chain id to auxiliary chain object.

Is the key a string or an integer? 🤔 

```suggestion
    public auxiliaryChains: {[key: string]: AuxiliaryChain};
```